ecs.tf

```diff
@@ -198,11 +199,24 @@ resource "aws_iam_role_policy" "secretsRead_policy" {
   policy = data.aws_iam_policy_document.secretsRead_policy_document.json
 }
 
+data "aws_kms_alias" "ssm_key" {
+  count = length(var.kms_key_aliases)
+  name = var.kms_key_aliases[count.index]
+}
+
 data "aws_iam_policy_document" "secretsRead_policy_document" {
   statement {
     actions = [
       "ssm:GetParameters",
     ]
-    resources = [var.datadog_api_key_from]
+    resources = concat([var.datadog_api_key_from], var.secrets[*].valueFrom)
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "kms:DescribeKey",
+      "kms:Decrypt"
```

Comment:
Does this allow you to decrypt ssm?

Reply:
I believe so. We have tested this with the PR in carbon-zero-terraform pointing to this branch, and it seems to be working, although I've not yet removed the db-password in the circle-ci config if the feature flag repo